### PR TITLE
Fe5 solo article view

### PR DESCRIPTION
**Pull request by tartebabba**
added a solo article view with url 'articles/:article_id'

### CODE REVIEW

File: src/utils/apis.js [hunk 1]
```diff
@@ -22,6 +21,17 @@ export function getArticles(params) {
     .catch((err) => console.log(err, 'all articles'));
 }
 
+export function getSingleArticle({article_id}) {
+  return axios
+  .get(`${baseURL}articles/${article_id}`)
+  .then(({ data }) => {
```
Comment (nc-saima-khanom) [inline src/utils/apis.js:27]: As we aren't processing the data in anyway here in the .then and just returning it, we don't need the .then(). just return the axios.get

File: src/utils/apis.js [hunk 2]
```diff
@@ -22,6 +21,17 @@ export function getArticles(params) {
...
+    return data;
+  })
+  .catch((err) => console.log(err, 'individual article'));
```
Comment (nc-saima-khanom) [inline src/utils/apis.js:30]: Nice way to make sure you know where the error is coming from by putting the string in the log of the error

File: src/components/IndividualArticle.jsx [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+    getSingleArticle(id).then(({ article }) => {
+      setCurrentArticle(article);
+      setIsLoading(false);
+    });
+  }, []);
+
+  if (isLoading) return <LoadingScreen />;
+
+  // console.log(currentArticle);
+  const {
```
Comment (nc-saima-khanom) [inline src/components/IndividualArticle.jsx:24]: Interesting!

File: src/components/IndividualArticle.jsx [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+
+export default function ArticleView() {
+  const [currentArticle, setCurrentArticle] = useState({});
+  const [isLoading, setIsLoading] = useState(false);
+
+  const id = useParams();
+
+  useEffect(() => {
+    setIsLoading(true);
+    getSingleArticle(id).then(({ article }) => {
```
Comment (nc-saima-khanom) [inline src/components/IndividualArticle.jsx:15]: nice!

File: src/components/IndividualArticle.jsx [hunk 5]
```diff
@@ -0,0 +1,48 @@
+import LoadingScreen from './Screens';
+import { useState, useEffect } from 'react';
+import { getSingleArticle } from '../utils/apis';
+import { useParams } from 'react-router-dom';
+
+export default function ArticleView() {
+  const [currentArticle, setCurrentArticle] = useState({});
+  const [isLoading, setIsLoading] = useState(false);
+
+  const id = useParams();
```
Comment (nc-saima-khanom) [inline src/components/IndividualArticle.jsx:11]: id should be destructured off the params object that is returned from the useParams

**Comment by nc-rose-mullan** [comment]
Hi Ted, thanks for sending this through. I've got a couple of points of feedback for you:
1. Your loading state in Articles.jsx - it's currently not performing as it should. Have a think about when you're setting isLoading to true or false. Do we want it to be set to false to begin with?

**Pull request merged**